src/services/codefixes/importFixes.ts

```diff
@@ -786,18 +790,34 @@ namespace ts.codefix {
 
     export function forEachExternalModuleToImportFrom(
         program: Program,
+        autoImportProvider: Program | undefined,
         host: LanguageServiceHost,
         from: SourceFile,
         filterByPackageJson: boolean,
-        cb: (module: Symbol) => void,
+        cb: (module: Symbol, moduleFile: SourceFile | undefined, program: Program) => void,
+    ) {
+        forEachExternalModuleToImportFromInProgram(program, host, from, filterByPackageJson, (module, file) => cb(module, file, program));
+        if (autoImportProvider) {
+            const start = timestamp();
+            forEachExternalModuleToImportFromInProgram(autoImportProvider, host, from, filterByPackageJson, (module, file) => cb(module, file, autoImportProvider));
+            host.log?.(`forEachExternalModuleToImportFrom autoImportProvider: ${timestamp() - start}`);
+        }

```

Comment:
There were surprisingly few changes necessary to the auto-import logic itself 🤷‍♂️